SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/activities/EditSneakerEntryActivity.java

```diff
@@ -0,0 +1,145 @@
+		mCategory = (EditText) findViewById(R.id.txtSneakerCategory);
+		mSaveButton = (Button) findViewById(R.id.btnSaveData);
+		mName = (EditText) findViewById(R.id.txtSneakerName);
+		mBrand = (EditText) findViewById(R.id.txtSneakerBrand);
+		mRarity = (EditText) findViewById(R.id.txtSneakerRarity);
+		mSellVal = (EditText) findViewById(R.id.txtSellingValue);
+		mDesc = (EditText) findViewById(R.id.txtSneakerDescription);
+		mThumbnail = (ImageView) findViewById(R.id.imgSneakerEntry);
+	}
+
+	private void saveSneaker() {
+		SneakerCategory category = new SneakerCategory();
+		category.setName(mCategory.getText().toString());
+		Sneaker sneaker = new Sneaker(mName.getText().toString(), mBrand
+				.getText().toString(), mRarity.getText().toString(), mSellVal
+				.getText().toString(), mDesc.getText().toString(), category, 0);
+
+		SneakerDirectory.get(this).addSneaker(sneaker);
+	}
+
+	private void setSneakerValues(Sneaker sneaker) {
+		sneaker.setName(mName.getText().toString());
+		sneaker.setBrand(mBrand.getText().toString());
+		sneaker.setSellingValue(mSellVal.getText().toString());
+		sneaker.setRarity(mRarity.getText().toString());
+		sneaker.setDescription(mDesc.getText().toString());
+		sneaker.setCategory(mCategory.getText().toString());
+	}
+
+	@SuppressLint("NewApi")
+	private void setupEditView() {
+		final boolean isFromDetailActivity = getIntent().getBooleanExtra(
+				SneakerDetailsActivity.DETAILS_FROMDETAILS_EXTRA, false);
+		final boolean isFromListActivity = getIntent().getBooleanExtra(
+				SneakerEntryListActivity.LIST_FROMLIST_EXTRA, false);
+
+		Bundle mBundle = new Bundle();
+		mBundle = getIntent().getExtras();
+		mSneaker = (Sneaker) mBundle.getSerializable("editsneaker");
+		id = mSneaker.getId().toString();
+
+		if (isFromDetailActivity || isFromListActivity) {
+			Drawable imgDrawable = ContextCompat.getDrawable(
+					getApplicationContext(), mSneaker.getThumbnailId());
+			mThumbnail.setImageDrawable(imgDrawable);
+
+		}
+		mName.setText(mSneaker.getName());
+		mBrand.setText(mSneaker.getBrand());
+		mSellVal.setText(mSneaker.getSellingValue());
+		mRarity.setText(mSneaker.getRarity());
+		mDesc.setText(mSneaker.getDescription());
+		mCategory.setText(mSneaker.getCategory().getName().toString());
+
+		mSaveButton = (Button) findViewById(R.id.btnSaveData);
+		mSaveButton.setOnClickListener(new View.OnClickListener() {
+
+			@Override
+			public void onClick(View v) {
+				if (isFromDetailActivity || isFromListActivity) {
+					ArrayList<Sneaker> s = SneakerDirectory.get(
+							getApplicationContext()).getAllSneakers();
+					for (Sneaker ss : s) {
+						if (id.equals(ss.getId().toString())) {
+							setSneakerValues(ss);
+							SneakerDirectory.get(getApplicationContext())
+									.saveSneakers();
+							Intent i = new Intent();
+							Bundle sneakerBundle = new Bundle();
+							sneakerBundle.putSerializable("updatedsneaker", ss);
+							i.putExtra("updatedsneaker", sneakerBundle);
```

Comment:
use a public static final string for the "updatedsneaker" keys. put it in this class and have other activities to statically get it from here